### PR TITLE
Bugfix/1484 unordered map fixes

### PR DESCRIPTION
**Pull request by carlosdiez**
This changes the return value of `erase` method of `unordered_map` from `void` to `iterator`

### CODE REVIEW

File: runtests.py [hunk 1]
```diff
@@ -423,6 +423,7 @@ def build_extension(self, ext):
                 compiler_obj = self.compiler
             if ext.language == 'c++':
                 compiler_obj.compiler_so.remove('-Wstrict-prototypes')
+                compiler_obj.compiler_so.append('-std=c++11')
```
Comment (robertwb) [inline runtests.py]: We don't want to change all our tests to require this. Instead, put it as a directive in the file in question.

**Comment by robertwb** [comment]
Looks like some tests failed.

File: runtests.py (hunk 1 above)
Comment (carlosdiez) [inline runtests.py]: Done, yes, much better to enable it just for this test. Thanks.

**Comment by carlosdiez** [comment]
Which compiler is being used on the build server?
Looks like it's not C++11 compatible
https://travis-ci.org/cython/cython/jobs/168405515#L7787

**Comment by carlosdiez** [comment]
Is it ok to update `.travis.yml` to use GCC 5.X?
http://genbattle.bitbucket.org/blog/2016/01/17/c++-travis-ci/

**Comment by carlosdiez** [comment]
Moved the tests to `broken/libcpp_11_all.pyx`, now the build is green.
`libcpp_11_all.pyx` can be moved to `run` if it's added into the test suite or not depending on the compiler version.

For example, the `tag` can be set to `cpp11` in here, and `runpython.py` will only execute this tests if the compiler supports it.

**Comment by MichaelWS** [comment]
Can we merge this in?

**Comment by MichaelWS** [comment]
Is there anything I can do to help get this merged?

**Comment by scoder** [comment]
The test is a) not really testing anything and b) moved to `broken/`, which disables it completely.
Things to do:
- move test file to `tests/run/`.
- move the test code into a function and add a doctest to it that calls and tests it.
- enable C++11 for this test, preferably with a new tag like `c++11` in `runtests.py` that sets appropriate compiler flags (and maybe also disables the test on unsupported platforms).

**Comment by MichaelWS** [comment]
thanks for the patience and the reply. this test does nothing, should i
throw some asserts in there?

at work, we use cython so it would be great if we can help.

I may need help with point 3.  It's a compiler issue vs a compiler flag.
gcc 5, 6, 7 will use c++11 if you add tag --std=c++0x
Thanks again

**Pull request closed without merging**